src/transport/raw/tests/TestTCP.cpp

```diff
@@ -205,10 +231,93 @@ class MockTransportMgrDelegate : public chip::TransportMgrDelegate
         SetCallback(nullptr);
     }
 
+    void MultipleConnectionTest(TCPImpl & tcp, const IPAddress & addr, uint16_t port)
+    {
+        ActiveTCPConnectionHolder firstConnection;
+        ActiveTCPConnectionHolder secondConnection;
+        bool firstMessageReceived   = false;
+        bool firstResponseReceived  = false;
+        bool secondMessageReceived  = false;
+        bool secondResponseReceived = false;
+        SetCallback(
+            [&](const uint8_t * message, size_t length, int count, ActiveTCPConnectionHolder & conn, void * data) -> CHIP_ERROR {
+                if (memcmp(message, PAYLOAD, length) == 0)
+                {
+                    firstMessageReceived = true;
+
+                    chip::System::PacketBufferHandle firstResponse;
+                    ReturnErrorOnFailure(BufferWithHeader(kMessageCounter + 2, firstResponse, PAYLOAD_RESPONSE));
+
+                    // First message, respond with first response
+                    ReturnErrorOnFailure(tcp.SendMessage(conn, std::move(firstResponse)));
```

Comment:
So, in the test we use this `SendMessage` which allows us to pass in a respective connection; but `SessionManager` can't use this, as it's not exposed by `mTransportMgr`

The reason we need it here is because this is a loop-back, and if we didn't specify the connection, we would use the wrong one

Do we care about and want to support the loopback scenario in `SessionManager`?